.github/workflows/release-cycle.yml

```
@@ -0,0 +1,34 @@
+name: Test release cycle
+
+on: [pull_request, push, workflow_dispatch]
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Silence Windows warnings
+        run: |
+          git config core.eol lf
+          git config core.autocrlf true
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3"
+
+      - name: Generate release cycle output
+        run: python _tools/generate_release_cycle.py
+
+      - name: Check for differences
+        run: |
+          git status --porcelain
```

```suggestion
          git status
          git diff
```

Maybe output the full status, as well as the actual diff? This would make debugging issues (like this one) much easier, no?